dra-evolution/pkg/api/capacity_types.go

```diff
@@ -64,12 +64,33 @@ type ResourcePoolSpec struct {
 	// Must not have more than 128 entries.
 	Devices []Device `json:"devices,omitempty"`
 
+	// SharedCapacity are pooled resources that are shared by all
+	// devices in the pool. This is typically used when representing a
```

Comment:
Is this the pool as defined in https://github.com/kubernetes/enhancements/pull/4709 or the object?

Reply:
This field would be directly in the ResourceSlice.

Reply:
This becomes relevant when considering a future "reservation" protocol. Reserving a device and the corresponding shared capacity will be a lot easier if in the same object.

Reply:
You don't have to reserve the shared capacity -- just the device (as before). By reserving a device, you implicitly decrease the amount of resources available in the shared capacity.

Reply:
I was thinking towards a future protocol for allocation when there are multiple schedulers in a cluster which need to allocate the same devices.

Reply:
"in the pool" should read "in the ResourceSlice", so they are the same object, yes!

You get a whole device or not, that is what determines the subtraction from the shared set. It doesn't matter if there are multiple schedulers or not; they just need to coordinate on the *device* allocation.

Reply:
Hmm, yes, then they somehow need to know how much of the overall shared capacity has already been depleted.

Reply:
Actually, so long as we can see the list of all devices that have already been `reserved`, then each scheduler can do the math to track the depleted capacity independently.

Reply:
yes, that's exactly how we do it, this is the same with and without shared resources.